plugins/obs-webrtc/whip-output.cpp

```diff
@@ -347,28 +349,55 @@ bool WHIPOutput::Connect()
 		return false;
 	}
 
-	if (location_header.empty()) {
+	long redirect_count = 0;
+	curl_easy_getinfo(c, CURLINFO_REDIRECT_COUNT, &redirect_count);
+
+	if (location_headers.size() < static_cast<size_t>(redirect_count) + 1) {
 		do_log(LOG_WARNING,
 		       "WHIP server did not provide a resource URL via the Location header");
-	} else {
-		CURLU *h = curl_url();
-		curl_url_set(h, CURLUPART_URL, endpoint_url.c_str(), 0);
-		curl_url_set(h, CURLUPART_URL, location_header.c_str(), 0);
-		char *url = nullptr;
-		CURLUcode rc = curl_url_get(h, CURLUPART_URL, &url,
-					    CURLU_NO_DEFAULT_PORT);
-		if (!rc) {
-			resource_url = url;
-			curl_free(url);
-			do_log(LOG_DEBUG, "WHIP Resource URL is: %s",
-			       resource_url.c_str());
-		} else {
-			do_log(LOG_WARNING,
-			       "Unable to process resource URL response");
+		cleanup();
+		obs_output_signal_stop(output, OBS_OUTPUT_CONNECT_FAILED);
+		return false;
+	}
+
+	CURLU *h = curl_url();
```

Comment:
Nit-picky but can we give this a more descriptive name? my head keeps reading the h as header further down, heh

Reply:
Done! Changed name to `url_builder` I think that better communicates its purpose.